src/app/checklist/checklist-search/checklist-search.component.ts

```diff
@@ -1,47 +1,43 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, computed, inject, signal } from '@angular/core';
 import { FormControl, ReactiveFormsModule } from '@angular/forms';
 import { MatAutocompleteSelectedEvent, MatAutocompleteTrigger, MatAutocomplete } from '@angular/material/autocomplete';
 import { Router } from '@angular/router';
 import * as fuzzysort from 'fuzzysort';
-import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
-import { debounceTime, map, switchMap } from 'rxjs/operators';
+import { debounceTime } from 'rxjs/operators';
 import { CategoryEntity, ChecklistItem } from '../models/checklist.model';
 import { IndexEntry, SearchResult } from '../search/search.models';
 import { SearchService } from '../search/search.service';
 import { MatOption } from '@angular/material/core';
-import { NgFor, NgIf, AsyncPipe } from '@angular/common';
+import { NgFor, NgIf } from '@angular/common';
+import { toSignal } from '@angular/core/rxjs-interop';
 
 @Component({
   standalone: true,
   selector: 'ac-checklist-search',
   templateUrl: './checklist-search.component.html',
   styleUrls: ['./checklist-search.component.scss'],
-  imports: [ReactiveFormsModule, MatAutocompleteTrigger, MatAutocomplete, NgFor, MatOption, NgIf, AsyncPipe]
+  imports: [ReactiveFormsModule, MatAutocompleteTrigger, MatAutocomplete, NgFor, MatOption, NgIf]
 })
-export class ChecklistSearchComponent implements OnInit {
-  results$: Observable<any>;
+export class ChecklistSearchComponent {
+  private searchService = inject(SearchService);
+  private router = inject(Router);
   searchField = new FormControl<string>('');
-
-  focus$ = new BehaviorSubject<string>('INIT');
-
-  constructor(private searchService: SearchService, private router: Router) {}
-
-  ngOnInit() {
-    const search$ = this.searchField.valueChanges.pipe(debounceTime(150));
-
-    this.results$ = combineLatest([this.focus$, search$]).pipe(
-      map(([, term]) => term),
-      switchMap(term => this.searchService.search(term)),
-      map(results => results.map(this.mapToSearchResult))
-    );
+  search = toSignal(this.searchField.valueChanges.pipe(debounceTime(150)));
+  focus = signal(0);
```

Comment:
I'd prefer to keep the 'INIT'  / 'FOCUS' logic as more explicit than using numbers.

Reply:
@geromegrignon we can't use 'INIT' / 'FOCUS' with signal because once we set the 'FOCUS' then on next focus when again set it as 'FOCUS' then signal won't trigger due to it equality check.